src/lorawan_mac.erl

```diff
@@ -125,17 +125,19 @@ check_link(DevAddr, FCnt) ->
         [L] ->
             case fcnt_gap(L#link.fcntup, FCnt) of
                 N when N < ?MAX_FCNT_GAP ->
-                    mnesia:dirty_write(links, L#link{fcntup=FCnt}),
+                    NewFCnt = (L#link.fcntup + N) band 16#FFFFFFFF,
```

Comment:
`L#link.fcntup` is 32bits, and `FCnt` is 16bits. It means that if we want to get the real `UpLinkCounter`, we must compare the **LSB** 16bits of the `L#link.fcntup` with `FCnt` : 
>the server must infer the 16 most-significant bits of the frame counter from the observation of the traffic

And we should use `L#link.fcntup + N` (or `(L#link.fcntup band 16#FFFF) band FCnt`) instead of using 16bits `FCnt` directly.